File: support_tools/swigwin-3.0.12/Examples/test-suite/python/overload_simple_runme.py
from overload_simple import *

if foo(3) != "foo:int":
    raise RuntimeError, "foo(int)"

if foo(3.0) != "foo:double":
    raise RuntimeError, "foo(double)"

if foo("hello") != "foo:char *":
    raise RuntimeError, "foo(char *)"

f = Foo()
b = Bar()

if foo(f) != "foo:Foo *":
    raise RuntimeError, "foo(Foo *)"

if foo(b) != "foo:Bar *":
    raise RuntimeError, "foo(Bar *)"

v = malloc_void(32)

if foo(v) != "foo:void *":
    raise RuntimeError, "foo(void *)"

s = Spam()

if s.foo(3) != "foo:int":
    raise RuntimeError, "Spam::foo(int)"

if s.foo(3.0) != "foo:double":
    raise RuntimeError, "Spam::foo(double)"

if s.foo("hello") != "foo:char *":
    raise RuntimeError, "Spam::foo(char *)"

if s.foo(f) != "foo:Foo *":
    raise RuntimeError, "Spam::foo(Foo *)"

if s.foo(b) != "foo:Bar *":
    raise RuntimeError, "Spam::foo(Bar *)"

if s.foo(v) != "foo:void *":
    raise RuntimeError, "Spam::foo(void *)"

if Spam_bar(3) != "bar:int":
    raise RuntimeError, "Spam::bar(int)"

if Spam_bar(3.0) != "bar:double":
    raise RuntimeError, "Spam::bar(double)"

if Spam_bar("hello") != "bar:char *":
    raise RuntimeError, "Spam::bar(char *)"

if Spam_bar(f) != "bar:Foo *":
    raise RuntimeError, "Spam::bar(Foo *)"

if Spam_bar(b) != "bar:Bar *":
    raise RuntimeError, "Spam::bar(Bar *)"

if Spam_bar(v) != "bar:void *":
    raise RuntimeError, "Spam::bar(void *)"

# Test constructors

s = Spam()
if s.type != "none":
    raise RuntimeError, "Spam()"

s = Spam(3)
if s.type != "int":
    raise RuntimeError, "Spam(int)"

s = Spam(3.4)
if s.type != "double":
    raise RuntimeError, "Spam(double)"

s = Spam("hello")
if s.type != "char *":
    raise RuntimeError, "Spam(char *)"

s = Spam(f)
if s.type != "Foo *":
    raise RuntimeError, "Spam(Foo *)"

s = Spam(b)
if s.type != "Bar *":
    raise RuntimeError, "Spam(Bar *)"

s = Spam(v)
if s.type != "void *":
    raise RuntimeError, "Spam(void *)"


free_void(v)


a = ClassA()
b = a.method1(1)

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/overload_simple_runme.go
package main

import . "./overload_simple"

func main() {
	if Foo(3) != "foo:int" {
		panic("foo(int)")
	}

	if Foo(3.0) != "foo:double" {
		panic("foo(double)")
	}

	if Foo("hello") != "foo:char *" {
		panic("foo(char *)")
	}

	f := NewFoos()
	b := NewBar()

	if Foo(f) != "foo:Foo *" {
		panic("foo(Foo *)")
	}

	if Foo(b) != "foo:Bar *" {
		panic("foo(Bar *)")
	}

	v := Malloc_void(32)

	if Foo(v) != "foo:void *" {
		panic("foo(void *)")
	}
	s := NewSpam()

	if s.Foo(3) != "foo:int" {
		panic("Spam::foo(int)")
	}

	if s.Foo(3.0) != "foo:double" {
		panic("Spam::foo(double)")
	}

	if s.Foo("hello") != "foo:char *" {
		panic("Spam::foo(char *)")
	}

	if s.Foo(f) != "foo:Foo *" {
		panic("Spam::foo(Foo *)")
	}

	if s.Foo(b) != "foo:Bar *" {
		panic("Spam::foo(Bar *)")
	}

	if s.Foo(v) != "foo:void *" {
		panic("Spam::foo(void *)")
	}

	if SpamBar(3) != "bar:int" {
		panic("Spam::bar(int)")
	}

	if SpamBar(3.0) != "bar:double" {
		panic("Spam::bar(double)")
	}

	if SpamBar("hello") != "bar:char *" {
		panic("Spam::bar(char *)")
	}

	if SpamBar(f) != "bar:Foo *" {
		panic("Spam::bar(Foo *)")
	}

	if SpamBar(b) != "bar:Bar *" {
		panic("Spam::bar(Bar *)")
	}

	if SpamBar(v) != "bar:void *" {
		panic("Spam::bar(void *)")
	}

	// Test constructors

	s = NewSpam()
	if s.GetXtype() != "none" {
		panic("Spam()")
	}

	s = NewSpam(3)
	if s.GetXtype() != "int" {
		panic("Spam(int)")
	}

	s = NewSpam(3.4)
	if s.GetXtype() != "double" {
		panic("Spam(double)")
	}

	s = NewSpam("hello")
	if s.GetXtype() != "char *" {
		panic("Spam(char *)")
	}

	s = NewSpam(f)
	if s.GetXtype() != "Foo *" {
		panic("Spam(Foo *)")
	}

	s = NewSpam(b)
	if s.GetXtype() != "Bar *" {
		panic("Spam(Bar *)")
	}

	s = NewSpam(v)
	if s.GetXtype() != "void *" {
		panic("Spam(void *)")
	}

	Free_void(v)

	a := NewClassA()
	_ = a.Method1(1)
}
